operator/pkg/controller/clusterconfig/virtualippool.go

```diff
@@ -0,0 +1,197 @@
+		if changed {
+			r.setVIPConfiguredCondition(
+				metav1.ConditionFalse,
+				"NamespaceDeletionInProgress",
+				fmt.Sprintf("No pools defined, deletion of the '%s' namespace in progress", vipNamespaceName),
+			)
+			return utils.NeedEndReconciliation()
+		}
+
+		r.setVIPConfiguredCondition(metav1.ConditionTrue, "NoPools", "Nothing to do, no pools defined")
+		return utils.NothingToDo()
+	}
+
+	// Create or update the namespace if needed
+	// NOTE: We treat the namespace alone has nothing else can be created if this one
+	// do not exists
+	changed, err := handler.CreateOrUpdateOrDelete(ctx, []client.Object{namespaceInstance}, nil, nil)
+	if err != nil {
+		r.setVIPConfiguredCondition(
+			metav1.ConditionUnknown,
+			"NamespaceUpdateError",
+			err.Error(),
+		)
+		return utils.NeedRequeue(err)
+	}
+	if changed {
+		r.setVIPConfiguredCondition(
+			metav1.ConditionFalse,
+			"NamespaceUpdateInProgress",
+			fmt.Sprintf("Creation/Update of the '%s' namespace in progress", vipNamespaceName),
+		)
+		return utils.NeedEndReconciliation()
+	}
+
+	// Retrieve all objects that should be created/updated
+	objsToUpdate := []client.Object{}
+	for name := range pools {
+		objsToUpdate = append(objsToUpdate, &metalk8sscalitycomv1alpha1.VirtualIPPool{
+			ObjectMeta: r.getPoolMeta(name),
+		})
+	}
+
+	// Retrieve all objects that should be deleted
+	objsToDelete := []client.Object{}
+	poolList := metalk8sscalitycomv1alpha1.VirtualIPPoolList{}
+	if err := r.client.List(ctx, &poolList, handler.GetMatchingLabels(false), client.InNamespace(vipNamespaceName)); err != nil {
+		return utils.NeedRequeue(err)
+	}
+	for _, obj := range poolList.Items {
+		if _, known := pools[obj.GetName()]; !known {
+			objsToDelete = append(objsToDelete, &obj)
+		}
+	}
+
+	changed, err = handler.CreateOrUpdateOrDelete(ctx, objsToUpdate, objsToDelete, r.mutateVIP)
+	if err != nil {
+		r.setVIPConfiguredCondition(
+			metav1.ConditionUnknown,
+			"ObjectUpdateError",
+			err.Error(),
+		)
+		return utils.NeedRequeue(err)
+	}
+	if changed {
+		r.setVIPConfiguredCondition(
+			metav1.ConditionFalse,
+			"ObjectUpdateInProgress",
+			"Creation/Update of various objects in progress",
+		)
+		return utils.NeedEndReconciliation()
+	}
+
+	r.setVIPConfiguredCondition(metav1.ConditionTrue, "Configured", "All objects properly configured")
+
+	// Wait for all VirtualIPPools to be Ready
+	// NOTE: We only have one global status for the whole ClusterConfig
+	// so we return on the first not ready pool
+	for name := range pools {
+		pool := &metalk8sscalitycomv1alpha1.VirtualIPPool{
+			ObjectMeta: r.getPoolMeta(name),
+		}
+		if err := r.client.Get(ctx, client.ObjectKeyFromObject(pool), pool); err != nil {
+			status := metav1.ConditionUnknown
+			reason := "PoolRetrievingError"
+			message := err.Error()
+			if errors.IsNotFound(err) {
+				status = metav1.ConditionFalse
+				reason = "PoolNotCreated"
+				message = fmt.Sprintf("The VirtualIPPool %s does not exists yet", pool.GetName())
```

Comment:
Since it's user-facing:

```suggestion
				message = fmt.Sprintf("The VirtualIPPool %s does not exist yet", pool.GetName())
```

Reply:
Done